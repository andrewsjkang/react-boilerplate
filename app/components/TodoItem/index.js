import React from 'react';
import { shape, string, number } from 'prop-types';

const TodoItem = props => (
  <div>
    <h2>{props.todo.task}</h2>
  </div>
);

TodoItem.propTypes = {
  todo: shape({
    id: number,
    task: string,
  }),
};

export default TodoItem;
