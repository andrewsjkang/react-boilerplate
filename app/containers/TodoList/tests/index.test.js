import React from 'react';
import { shallow } from 'enzyme';

import { TodoList } from '../index';
import TodoItem from '../../../components/TodoItem';
import Spinner from '../../../components/Spinner';

describe('<TodoList />', () => {
  const disableLifecycleMethods = { disableLifecycleMethods: true };

  it('should render properly', () => {
    const component = shallow(<TodoList todos={[]} />, disableLifecycleMethods);
    expect(component).toMatchSnapshot();
  });

  it('should render correct amount of todos', () => {
    const testData = [{ id: 1, task: 'foo' }, { id: 2, task: 'bar' }];
    const component = shallow(
      <TodoList todos={testData} />,
      disableLifecycleMethods,
    );
    expect(component.find(TodoItem).length).toEqual(testData.length);
  });

  it('should render Spinner when loading', () => {
    const component = shallow(<TodoList loading />, disableLifecycleMethods);
    expect(component.contains(<Spinner />)).toEqual(true);
  });
});
