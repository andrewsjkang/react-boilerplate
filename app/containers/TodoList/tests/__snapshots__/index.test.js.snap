// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render properly 1`] = `
<div>
  <h1>
    Todo List
  </h1>
</div>
`;
